modoboa/admin/models/domain.py

```diff
@@ -224,6 +224,17 @@ def post_create(self, creator):
 reversion.register(Domain)
 
 
+class MXQuerySet(models.QuerySet):
+    """Custom manager for MXRecord."""
+
+    def valids(self):
+        """Return managed results."""
+        if parameters.get_admin("ENABLE_MX_CHECKS"):
+            if parameters.get_admin("VALID_MXS", app="admin").strip():
+                return self.filter(managed=True).exists()
+        return self.exists()
```

Comment:
So you just check is there is manager MXs in the database ? You don't want to retrieve them with this method ?